src/commands/init.ts

```diff
@@ -52,183 +53,128 @@ export default class Init extends Command {
+      buildScaffold ? this.printSuccessMessageScaffold() : this.printSuccessMessage();
     } catch (error) {
       this.error(error, { exit: 1 });
     }
   }
 
-  private async determineInitializationType(flags: Flags): Promise<InitType> {
-    if (fs.existsSync(path.resolve(flags.projectDir, 'package.json'))) {
-      return InitType.existingProject;
+  private async shouldBuildScaffold(projectDir: string, buildScaffold: boolean) {
+    if (buildScaffold) {
+      return true;
     }
 
-    this.log('This folder does not contain a `package.json`.');
-    if (flags.initWithExpress || (await cli.confirm('Should a new `express.js` project be initialized in this folder?'))) {
-      return InitType.freshExpress;
+    if (fs.existsSync(path.resolve(projectDir, 'package.json'))) {
+      return false;
     }
 
-    return InitType.existingProject;
-  }
+    this.log('This folder does not contain a `package.json`.');
 
-  private async initProject(flags: Flags, initializationType: InitType): Promise<void> {
-    switch (initializationType) {
-      case InitType.freshExpress:
-        return this.initExpressProject(flags);
-      case InitType.existingProject:
-        return;
-    }
+    return cli.confirm('Should a new `nest.js` project be initialized in this folder?');
   }
 
-  private async initExpressProject(flags: Flags): Promise<void> {
-    cli.action.start('Initializing Express project');
-
-    const params = ['express-generator', '--no-view', '--git'];
-    const dirEmpty = fs.readdirSync(flags.projectDir).length === 0;
-
-    if (!dirEmpty && (flags.force || (await cli.confirm('Directory is not empty. Should the project be initialized anyway?')))) {
-      params.push('--force');
-    }
-    await execa('npx', params, { cwd: flags.projectDir });
-
-    if (!fs.existsSync('.git')) {
-      await execa('git', ['init'], { cwd: flags.projectDir });
+  private async buildScaffold(projectDir: string, { force, verbose }: Flags) {
+    if (fs.readdirSync(projectDir).length !== 0) {
+      const dirString = projectDir === '.' ? 'this directory' : projectDir;
+      if (force || (await cli.confirm(`Directory is not empty. Remove all files in ${dirString}?`))) {
+        rm.sync(`${projectDir}/{*,.*}`);
+      }
     }
-    cli.action.stop();
-  }
 
-  private async getOptions(flags: Flags) {
+    const cliPath = path.resolve('node_modules/.bin/nest');
     const options: { [key: string]: string } = {
-      projectName:
-        flags.projectName ||
-        (await cli.prompt('Enter project name (for use in manifest.yml)', {
-          default: this.packageJson(flags).name
-        })),
-      command:
-        flags.startCommand ||
-        (await cli.prompt('Enter the command to start your server', {
-          default: this.packageJson(flags).scripts.start ? 'npm start' : ''
-        }))
+      cwd: projectDir,
+      stdio: verbose ? 'inherit' : 'ignore'
     };
 
-    return options;
+    if (fs.existsSync(cliPath)) {
```

Comment:
This is probably not necessary for now, but if we do this in multiple places, we should probably abstract this away in some fashion.